drivers/painter/fallback/qp_fallback.c

```diff
@@ -95,3 +95,137 @@ bool qp_fallback_rect(painter_device_t device, uint16_t left, uint16_t top, uint
 
     return true;
 }
+
+// Fallback implementation for drawing circles
+bool qp_fallback_circle(painter_device_t device, uint16_t x, uint16_t y, uint16_t radius, uint8_t hue, uint8_t sat, uint8_t val, bool filled) {
+
+// plot the initial set of points for x, y and r
+    uint16_t xcalc, ycalc, err;
+    xcalc = 0;
+    ycalc = radius;
+    err = ((5 - (radius >> 2)) >> 2);
+
+    qp_fallback_circle_drawpixels(device, x, y, xcalc, ycalc, hue, sat, val, filled);
+    while (xcalc < ycalc) {
+        xcalc++;
+        if (err < 0) {
+            err += (xcalc << 1) + 1;
+        } else {
+            ycalc--;
+            err += ((xcalc - ycalc) << 1) + 1;
+        }
+        qp_fallback_circle_drawpixels(device, x, y, xcalc, ycalc, hue, sat, val, filled);
```

Comment:
This file will need a declaration for this function above this line -- it's lower in the file so isn't yet visible.

Reply:
added